docs/gcloud-auth.rst

```diff
@@ -0,0 +1,310 @@
+.. toctree::
+  :maxdepth: 1
+  :hidden:
+
+GCloud Auth
+-----------
+
+For the majority of cases, authentication with the ``gcloud`` library
+will **"just work"**. Credentials are loaded implicitly from the environment
+where the code is running and **no code** is required to enable authentication
+within your code.
+
+The core :mod:`gcloud.credentials` module provides helper methods
+for any account type needed.
+
+A production application should **use a service account**, but you may
+wish to use other methods when first getting started with the ``gcloud``
+library.
+
+=============
+Basic Example
+=============
+
+After downloading a JSON key for a service account (see `Service Accounts`_),
+set the ``GOOGLE_APPLICATION_CREDENTIALS`` environment variable:
+
+.. code-block:: bash
+
+  $ export GOOGLE_APPLICATION_CREDENTIALS="/path/to/key_file.json"
+
+After doing this, your API requests will be authenticated automatically
+using the JSON key:
+
+.. code-block:: python
+
+   from gcloud import datastore
+   from gcloud import pubsub
+   from gcloud import storage
+
+   key1 = datastore.Key('EntityKind', 1, dataset_id='foo')
+   key2 = datastore.Key('EntityKind', 2, dataset_id='foo')
+   entities = datastore.get([key1, key2])
+
+   bucket = storage.get_bucket('bucket-name')
+
+   topic = pubsub.Topic('topic_name', project_id='my.project')
+   topic.create()
+
+=============
+Advanced Uses
+=============
+
+A custom connection can be used as the default, but will need to be
+explicitly set. The
+:func:`set_default_connection() <gcloud.datastore.__init__.set_default_connection>`
+function is provided in all sub-packages to set a connection. For
+example, in the ``datastore`` sub-package:
+
+.. code-block:: python
+
+   credentials = get_custom_credentials()
+   connection = datastore.Connection(credentials=credentials)
+   datastore.set_default_connection(connection)
+
+   # ... Create keys.
+   entities = datastore.get([key1, key2])
+
+If no default is set, all functions which make a connection to an API
+accept an optional ``connection`` argument. For example, with ``datastore``,
+the implicit behavior can be duplicated with:
+
+.. code-block:: python
+
+   from gcloud.credentials import get_credentials
+
+   credentials = get_credentials().create_scoped(datastore.SCOPE)
+   connection = datastore.Connection(credentials=credentials)
```

Comment:
@jgeewax This relates to comments in #787.

We could alter the constructors of `Connection` subclasses so that this is equivalent to

```python
credentials = get_credentials()
connection = datastore.Connection(credentials=credentials)
```

but I don't think it is worthwhile to jam them into one line / spawn yet another method.